File: content/polympiads.2/4.alice-et-bob/.codes/empty.cpp
#include <iostream>
#include <vector>

using namespace std;

int main () {
    int nbArrets;
    cin >> nbArrets;

    vector<int> parents(nbArrets);
    for (int i = 0; i < nbArrets; i ++) {
        cin >> parents[i];
    }

    int nbQuestions;
    cin >> nbQuestions;

    vector<int> arretAlice (nbQuestions);
    vector<int> arretBob (nbQuestions);
    for (int i = 0; i < nbQuestions; i ++) {
        cin >> arretAlice[i] >> arretBob[i];
    }
    
    // Affichez pour chaque question le premier arrêt où Alice et Bob peuvent se rejoindre
}

File: content/polympiads.2/4.alice-et-bob/.codes/empty.py
nb_arrets = int(input())
parents = list(map(int, input().split()))
nb_questions = int(input())
arret_alice = []
arret_bob = []
for i in range(nb_questions):
    a, b = list(map(int, input().split()))
    arret_alice.append(a)
    arret_bob.append(b)

# Affichez pour chaque question le premier arrêt où Alice et Bob peuvent se rejoindre